dags/aggregates_python_helpers.py

```diff
@@ -143,13 +149,20 @@ def get_first_day_of_previous_month(date_str):
     new_date = datetime(year, month, 1)
```

Comment:
I am advice also check simpler solution. ChatGPT propose:
from datetime import datetime, timedelta

def get_first_day_of_previous_month(date_str):
    # Convert the input string to a datetime object
    current_date = datetime.strptime(date_str, '%Y-%m-%d %H:%M:%S')
    
    # Subtract one month from the current date
    previous_month_date = current_date.replace(day=1) - timedelta(days=1)
    
    # Get the first day of the previous month
    first_day_of_previous_month = previous_month_date.replace(day=1)
    
    # Format the result as a string in the same format as the input
    return first_day_of_previous_month.strftime('%Y-%m-%d %H:%M:%S')